reactor-core/src/main/java/reactor/core/publisher/FluxHandle.java

```diff
@@ -270,19 +307,31 @@ public void onNext(T t) {
 				actual.onNext(v);
 			}
 			if(done){
-				s.cancel();
 				if(error != null){
-					actual.onError(Operators.onOperatorError(null, error, t,
-							actual.currentContext()));
-					return;
+					Throwable e_ = Operators.onNextError(t, error, actual.currentContext(), s);
+					if (e_ != null) {
+						actual.onError(e_);
+						return;
+					}
+					else {
+						reset();
+						s.request(1L);
+					}
+				}
+				else {
+					actual.onComplete();
```

Comment:
need to do the `s.cancel()` in this block

Reply:
Cancellation on completion?

Reply:
the old code would always cancel `s`, whereas with your modification it only cancels when there is an error without continue mode, not when there is completion.